File: Data Structures & Algorithms/two-integer-sum-ii/submission-6.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> result = {};

        unordered_map<int, int> freq;
        for (int i = 0; i < numbers.size(); i++) {
            int first_num = numbers[i];

            int second_num = target - first_num;

            if (freq.count(second_num)){
                result = {freq[second_num], i + 1};
            }

            freq[first_num] = i + 1;
        }

        return result;

    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-6.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        result = 0

        for i in range(0, len(s)):
            v = set()
            for j in range(i, len(s)):
                if s[j] in v:
                    break
                v.add(s[j])
                result = max(result, len(v))

        return result

        